rust/worker/src/system/types.rs

```diff
@@ -135,6 +138,121 @@ where
     pub(crate) system: System,
     pub(crate) sender: Sender<C>,
     pub(crate) cancellation_token: tokio_util::sync::CancellationToken,
+    pub(crate) scheduler: Scheduler,
+}
+
+pub(crate) struct SchedulerHandle {
```

Comment:
Nit: Naming - How do you feel about ScheduledTaskHandle - SchedulerHandle feels like a handle to the scheduler